ThunderTable/TableViewController.swift

```diff
@@ -310,20 +310,40 @@ open class TableViewController: UITableViewController, UIContentSizeCategoryAdju
             size = imageSize
         }
         
-        imageView?.set(imageURL: row.imageURL, withPlaceholder: row.image, imageSize: size, animated: true, completion: { [weak self] (image, error) -> (Void) in
+        // Only load image from url if we have an image url
+        if let imageURL = row.imageURL {
             
-            if let welf = self, _row.image == nil {
+            imageView?.set(imageURL: imageURL, withPlaceholder: row.image, imageSize: size, animated: true, completion: { [weak self] (image, error) -> (Void) in
+                
+                guard let self = self else { return }
+                guard let image = image, _row.image == nil else { return }
                 _row.image = image
-                welf.tableView.reloadRows(at: [indexPath], with: .none)
-            }
-        })
+                // Try and find matching index paths with the same imageURL and reload it.
+                // We can't use `indexPath` provided to this function as it may have changed!
+                self.reloadRows(where: { $0.imageURL == imageURL })
+            })
+        }
 		
```

Comment:
@simonmitchell Do we need undo what `imageView?.set(imageURL:...)` might do when `row.imageURL` is not defined for reusable cells

Reply:
All it will do is set `imageView.image` that is undone on line `302` where the image is either set to `row.image` or `nil` :D